### ISSUE TITLE
Observed some data loss when kafka is not running and data is inserted into mysql

### ISSUE THREAD

**Issue by jaggukaka** [zendesk/maxwell#642]
Steps:
1. Stop kafka for few minutes (just to simulate a crash of kafka)
2. insert data into mysql

case 1: If maxwell receives the following message, there is data loss 
```
16:39:36,892 ERROR MaxwellKafkaProducer - TimeoutException @ BinlogPosition[master.000085:56123] -- {"database":"db","table":"abc","pk.id":"efg"}
16:39:36,892 ERROR MaxwellKafkaProducer - Batch Expired
```

case 2: If kafka is up before the error is thrown, there's no data loss.

Is this expected? is there any setting to overcome this data loss. The data loss is consistent in case1 and can be replicated with basic setup.

Also I tried following configuration for acks from kafka, but it didn't help
```
kafka.retries=5
kafka.acks=all
```

Please let me know if I'm missing some configuration.

**Comment by jaggukaka** [comment]
When I set min.insync.replicas and removed kafka.metadata.fetch.timeout.ms, looks like data loss issue is resolved, but I had to restart maxwell after every restart of Kafka. Is this required? is it possible for maxwell to auto re-connect to kafka and push data?

**Comment by jaggukaka** [comment]
Sorry, data loss is still there despite setting min.insync.replicas. So, whenever maxwell errors out with following error, there is data loss.
```
16:39:36,892 ERROR MaxwellKafkaProducer - TimeoutException @ BinlogPosition[master.000085:56123] -- {"database":"db","table":"abc","pk.id":"efg"}
16:39:36,892 ERROR MaxwellKafkaProducer - Batch Expired
```

**Comment by liulikun** [comment]
https://github.com/zendesk/maxwell/pull/635 is to solve this issue. Basically upon Kafka errors, maxwell can be terminated instead of just logging the error. You'll need to have some sort of monitoring for maxwell, so it can be auto restarted when it's down, this way once kafka is back up, maxwell will start processing again.

**Comment by jaggukaka** [comment]
Thanks!. Is the issue #635 fixed? if so is it available in the latest distribution or do I have to pull source code from master or some other branch and build?

**Comment by liulikun** [comment]
The PR is pending review, once it's merged we'll release a new version.

**Comment by jaggukaka** [comment]
Okay. Thanks!

**Comment by jaggukaka** [comment]
I pulled latest master and set the property ignore_producer_error=false in the config.properties. 
When maxwell errors out with following error, the the maxwell is terminated as expected

```
10:55:25,963 ERROR MaxwellKafkaProducer - Topic 'some_topic' name does not exist. Exception: Failed to update metadata after 5000 ms.
10:55:25,963 INFO  TaskManager - stopping 3 tasks
10:55:25,963 ERROR TaskManager - cause: 
org.apache.kafka.common.errors.TimeoutException: Failed to update metadata after 5000 ms.
10:55:25,965 INFO  StoppableTaskState - MaxwellKafkaProducerWorker requestStop() called (in state: STOPPED)
10:55:25,967 INFO  KafkaProducer - Closing the Kafka producer with timeoutMillis = 9223372036854775807 ms.
10:55:25,969 INFO  StoppableTaskState - com.zendesk.maxwell.schema.PositionStoreThread requestStop() called (in state: RUNNING)
10:55:25,970 INFO  StoppableTaskState - com.zendesk.maxwell.replication.MaxwellReplicator requestStop() called (in state: RUNNING)
10:55:26,170 INFO  TaskManager - stopped all tasks
org.apache.kafka.common.errors.TimeoutException: Failed to update metadata after 5000 ms.
```

But when the maxwell errors out with the following, maxwell is not terminated and the data for which the error is printed is always lost.

```
16:39:36,892 ERROR MaxwellKafkaProducer - TimeoutException @ BinlogPosition[master.000085:56123] -- {"database":"db","table":"abc","pk.id":"efg"}
16:39:36,892 ERROR MaxwellKafkaProducer - Batch Expired
```

Don't know if the above case has been missed while fixing #635. Just a data point from my observation, if it has been fixed already then great!

**Comment by jaggukaka** [comment]
Sorry, looks like your fix is in a different branch likun/ignore-producer-error yet to be merged, I misread the commit log. Will wait for the merge. Thanks!

**Comment by liulikun** [comment]
the fix is released in v1.10.1

**Comment by engineerashish007** [comment]
New to Maxwell,researching it to implement it while transferring bin-logs from MySql to Kafka.Please help me know certain things:
1)Can we configure the frequency at which we transfer data from source to Kafka? like every 1 min or every 5 minutes.
2)Do it support other database sources apart from MySql,I read somewhere it supports ProxySql.Please tell me if it supports MongoDB,Postgre or other tools.
